x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalyzeAction.java

```diff
@@ -444,12 +448,29 @@ private boolean isRunning() {
             return true;
         }
 
+        private class CheckForCancelListener implements ActionListener<Void> {
+            @Override
+            public void onResponse(Void unused) {
+                // task complete, nothing to do
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // trigger another isRunning check which will cancel the task if not already failed or cancelled
+                var isNowRunning = isRunning();
```

Comment:
Nit: I find it a bit surprising that `isRunning` does a bit more than an usualy `isXxx` boolean method. Not suggesting any change for this PR though since it is existing code and the name is not related to what we are trying to fix here.

Reply:
That's fair too. This was done before #82685 introduced `org.elasticsearch.tasks.CancellableTask#addListener`, but if we migrated to that we could drop the `task.isCancelled()` check here. I opened https://github.com/elastic/elasticsearch/issues/101197.